lib/compress/zstd_compress.c

```diff
@@ -2013,6 +2036,14 @@ static size_t ZSTD_resetCCtx_internal(ZSTD_CCtx* zc,
             zc->ldmState.loadedDictEnd = 0;
         }
 
+        /* reserve space for block-level external sequences */
+        if (params->useExternalMatchFinder) {
+            size_t const maxNbExternalSeq = ZSTD_sequenceBound(ZSTD_BLOCKSIZE_MAX);
```

Comment:
optimization : 
`blockSize` is `<= ZSTD_BLOCKSIZE_MAX`.
Prefer employing `blockSize` since it's available,
it will reduce memory budget.
This seems especially useful if the main use case is about small data.